### PR TITLE
[Function] Fix build status doesn't return the created image after reaching `ready` state 

### PR DESCRIPTION
**Pull request by Tankilevitch**
https://jira.iguazeng.com/browse/ML-3540

Aligning how we were returning the created image by the build flow, after build finished we are setting the `spec.build.image` to the `spec.image` and cleaning the `spec.build.image`, but we had a bug that in one of the flows when function was already ready we were returning the `spec.build.image` instead of the `spec.image`

### CODE REVIEW

**Review by theSaarco — APPROVED** [review]
Looks good as a temporary solution until we refactor the runtime spec/status fields.

File: mlrun/api/api/endpoints/functions.py [hunk 1]
```diff
@@ -458,7 +472,9 @@ def _handle_job_deploy_status(
         logger.error(f"build {state}, watch the build pod logs: {pod}")
         state = mlrun.api.schemas.FunctionState.error
 
-    if (logs and state != "pending") or state in terminal_states:
+    if (
+        logs and state != mlrun.api.schemas.FunctionState.pending
```
Comment (alonmr) [inline mlrun/api/api/endpoints/functions.py:476]: this state is not a `FunctionState` it is the pod's state

File: mlrun/api/api/endpoints/functions.py [hunk 2]
```diff
@@ -417,7 +417,12 @@ def _handle_job_deploy_status(
     image = get_in(fn, "spec.build.image", "")
     out = b""
     if not pod:
-        if state == "ready":
+        if state == mlrun.api.schemas.FunctionState.ready:
+            # when function has been built we set the created image in to the `spec.image` for reference see at the end
+            # of the function where we resolve if we in status ready and then set the spec.build.image to spec.image
+            # TODO: spec shouldn't hold backend enriched attributes, but rather in the status block
+            #   therefore need to change to set it to a new attribute in status.image which will ease on our resolution
+            #   of whether it is a user defined image or MLRun enriched one.
```
Comment (alonmr) [inline mlrun/api/api/endpoints/functions.py:425]:
```suggestion
            # when the function has been built we set the created image into the `spec.image` for reference see at the end
            # of the function where we resolve if the status is ready and then set the spec.build.image to spec.image
            # TODO: spec shouldn't hold backend enriched attributes, but rather in the status block
            #   therefore need set it as a new attribute in status.image which will ease our resolution
            #   of whether it is a user defined image or MLRun enriched one.
```

**Review by alonmr — COMMENTED** [review]
looks good, minor comments

**Review by alonmr — APPROVED** [review]

**Pull request merged**